### PR TITLE
Remove precommit write

### PR DESCRIPTION
**Pull request by triump2020**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [x] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #10272 

## What this PR does / why we need it:

### CODE REVIEW

File: pkg/txn/client/operator.go [hunk 1]
```diff
@@ -529,12 +529,14 @@ func (tc *txnOperator) doWrite(ctx context.Context, requests []txn.TxnRequest, c
 	if tc.option.readyOnly {
 		util.GetLogger().Fatal("can not write on ready only transaction")
 	}
-
+	var payload []*txn.TxnRequest
```
Comment (zhangxu19830126) [inline pkg/txn/client/operator.go:532]: donot use pointer. use value type

File: pkg/vm/engine/disttae/txn.go [hunk 2]
```diff
@@ -17,6 +17,7 @@ package disttae
 import (
 	"context"
 	"fmt"
+	"github.com/matrixorigin/matrixone/pkg/pb/txn"
```
Comment (zhangxu19830126) [inline pkg/vm/engine/disttae/txn.go:20]: format imports

File: pkg/txn/client/types.go [hunk 3]
```diff
@@ -183,7 +183,7 @@ type Workspace interface {
 	// Adjust adjust workspace, adjust update's delete+insert to correct order and merge workspace.
 	Adjust() error
 
-	Commit(ctx context.Context) error
+	Commit(ctx context.Context) ([]*txn.TxnRequest, error)
```
Comment (zhangxu19830126) [inline pkg/txn/client/types.go:186]: use value type slice

File: pkg/vm/engine/disttae/tools.go [hunk 4]
```diff
@@ -739,7 +739,7 @@ func newColumnExpr(pos int, typ *plan.Type, name string) *plan.Expr {
 	}
 }
 
-func genWriteReqs(ctx context.Context, writes []Entry) ([]txn.TxnRequest, error) {
+func genWriteReqs(ctx context.Context, writes []Entry) ([]*txn.TxnRequest, error) {
```
Comment (zhangxu19830126) [inline pkg/vm/engine/disttae/tools.go:742]: use value type slice

File: pkg/vm/engine/disttae/txn.go [hunk 5]
```diff
@@ -642,28 +643,27 @@ func (txn *Transaction) getCachedTable(
 	return nil
 }
 
-func (txn *Transaction) Commit(ctx context.Context) error {
+func (txn *Transaction) Commit(ctx context.Context) ([]*txn.TxnRequest, error) {
```
Comment (zhangxu19830126) [inline pkg/vm/engine/disttae/txn.go:646]: use value type slice

**Review by zhangxu19830126 — CHANGES_REQUESTED** [review]

**Review by daviszhen — APPROVED** [review]

**Review by nnsgmsone — APPROVED** [review]

**Review by qingxinhome — APPROVED** [review]

**Review by XuPeng-SH — APPROVED** [review]

**Pull request merged**